File: solution/solution-ecs-sts-token/code-example/python/sls_sdk/credentials_default_sample.py
from aliyun.log import LogClient
from aliyun.log.credentials import CredentialsProvider, Credentials

from alibabacloud_credentials.client import Client as CredentialsClient

class ECSRoleCredentialDemo(CredentialsProvider):
    def __init__(self, client):
        self.client = client

    def get_credentials(self):
        cred = self.client.get_credential()
        access_key_id = cred.get_access_key_id()
        access_key_secret = cred.get_access_key_secret()
        security_token = cred.get_security_token()
        return Credentials(access_key_id, access_key_secret, security_token)


# 默认凭据链方式初始化Credentials客户端
# 请确保Credentials Python SDK（alibabacloud-credentials）版本>=0.3.5
cred = CredentialsClient()

credentials_provider=ECSRoleCredentialDemo(cred)
client = LogClient("cn-hangzhou-intranet.log.aliyuncs.com", credentials_provider=credentials_provider)

response = client.list_project()
print(response.body)

File: solution/solution-ecs-sts-token/code-example/php/sls_sdk/credentials_default_sample.php
<?php
require 'vendor/autoload.php';
use AlibabaCloud\Credentials\Credential;

echo "Hello, World!";
class CredentialsAdapter implements Aliyun_Log_Models_CredentialsProvider
{
    /**
     * @var Credential
     */
    private $_credentials;
    public function __construct(Credential $credentials)
    {
        $this->_credentials = $credentials;
    }
    public function getCredentials(): Aliyun_Log_Models_Credentials
    {
        return new Aliyun_Log_Models_Credentials(
            $this->_credentials->getAccessKeyId(),
            $this->_credentials->getAccessKeySecret(),
            $this->_credentials->getSecurityToken()
        );
    }
};

# 设置参数
$endpoint = 'cn-hangzhou.log.aliyuncs.com';
$project = 'your-project';
$logstore = 'your-logstore';
$credentials = new Credential([
    'type' => 'ecs_ram_role',
    'role_name' => 'ecs-ak-role',
]);

# 构造 client
$credentialsProvider = new CredentialsAdapter($credentials);
$client = new Aliyun_Log_Client($endpoint, "", "", "", $credentialsProvider);


# 发请求
$req = new Aliyun_Log_Models_GetLogsRequest($project, $logstore, 1698740109, 1698744321, '', '*', null, null, null, null);
function putLogs(Aliyun_Log_Client $client, $project, $logstore) {
  $topic = 'TestTopic';

  $contents = array(
      'TestKey'=>'TestContent',
      'kv_json'=>'{"a": "b", "c": 19021}'
  );
  $logItem = new Aliyun_Log_Models_LogItem();
  $logItem->setTime(time());
  $logItem->setContents($contents);
  $logitems = array($logItem);
  $request = new Aliyun_Log_Models_PutLogsRequest($project, $logstore,
          $topic, null, $logitems);

  try {
      $response = $client->putLogs($request);
  } catch (Aliyun_Log_Exception $ex) {
      var_dump($ex);
  } catch (Exception $ex) {
      var_dump($ex);
  }
}

putLogs($client, $project, $logstore);
var_dump($client->getLogs($req)->getLogs());